### ISSUE TITLE
Pop Shop crashes shortly after launching

### ISSUE THREAD

**Issue by PGQT** [pop-os/shop#418]
**Distribution (run `cat /etc/os-release`):**
 
 NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**

io.elementary.appcenter

**Issue/Bug Description:**

Pop Shop freezes, then crashes shortly after launching. Typically within 15 seconds.

**Steps to reproduce (if you know):**

Launch pop shop

**Expected behavior:**

pop shop opens and allows the updating of existing apps and the searching and downloading of new apps

**Other Notes:**

I have tried restarting the system with no improvements noted.
I have run the follwing commands, which usually resolve any issues with updates, but with no improvements noted:
apt install --fix-missing
apt install --fix-broken
dpkg --configure -a

Launching from the CLI results in the following:

@pop-os:~$ io.elementary.appcenter 

(io.elementary.appcenter:4861): Gtk-WARNING **: 08:10:45.950: Theme parsing error: application.css:36:19: The :insensitive pseudo-class is deprecated. Use :disabled instead.
(io.elementary.appcenter:4861): GLib-CRITICAL **: 08:10:51.065: g_strsplit: assertion 'string != NULL' failed
Segmentation fault (core dumped)

/var/log/syslog:

Sep 21 08:35:17 pop-os gnome-shell[2152]: meta_window_set_stack_position_no_sync: assertion 'window->stack_position >= 0' failed
Sep 21 08:35:17 pop-os gnome-shell[2152]: message repeated 2 times: [ meta_window_set_stack_position_no_sync: assertion 'window->stack_position >= 0' failed]
Sep 21 08:35:17 pop-os PackageKit: resolve transaction /4933_bbadaead from uid 1000 finished with success after 302ms
Sep 21 08:35:18 pop-os PackageKit: get-details transaction /4934_eadecdbd from uid 1000 finished with success after 282ms
Sep 21 08:35:18 pop-os PackageKit: get-packages transaction /4935_ddaceaed from uid 1000 finished with success after 425ms
Sep 21 08:35:18 pop-os kernel: [ 1633.126827] io.elementary.a[32200]: segfault at 0 ip 000055fc1318a038 sp 00007ffceea75020 error 4 in io.elementary.appcenter[55fc13169000+68000] likely on CPU 4 (core 0, socket 0)
Sep 21 08:35:18 pop-os kernel: [ 1633.126839] Code: 41 5e e9 4b 41 fe ff 0f 1f 00 48 8b 3c 25 00 00 00 00 e8 0b 24 fe ff 4c 89 ef 49 89 c4 e8 d0 23 fe ff e9 f2 fe ff ff 0f 1f 00 <48> 8b 3c 25 00 00 00 00 e8 eb 23 fe ff 4c 89 e7 49 89 c5 e8 b0 23
Sep 21 08:35:18 pop-os PackageKit: resolve transaction /4936_ddadcabd from uid 1000 finished with success after 247ms

apt update:

Hit:1 http://apt.pop-os.org/proprietary jammy InRelease
Hit:2 http://apt.pop-os.org/release jammy InRelease
Hit:3 http://apt.pop-os.org/ubuntu jammy InRelease
Get:4 http://apt.pop-os.org/ubuntu jammy-security InRelease [110 kB]
Get:5 http://apt.pop-os.org/ubuntu jammy-updates InRelease [119 kB]
Get:6 https://repo.waydro.id jammy InRelease [1,309 B]  
Err:6 https://repo.waydro.id jammy InRelease               
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 0E406D181DCEE19C
Get:7 http://apt.pop-os.org/ubuntu jammy-backports InRelease [109 kB]
Get:8 http://apt.pop-os.org/ubuntu jammy-security/universe Sources [156 kB]
Get:9 http://apt.pop-os.org/ubuntu jammy-security/main amd64 DEP-11 Metadata [43.0 kB]
Get:10 http://apt.pop-os.org/ubuntu jammy-security/universe i386 Packages [559 kB]
Get:11 http://apt.pop-os.org/ubuntu jammy-security/universe amd64 Packages [785 kB]
Get:12 http://apt.pop-os.org/ubuntu jammy-security/universe Translation-en [143 kB]
Get:13 http://apt.pop-os.org/ubuntu jammy-security/universe amd64 DEP-11 Metadata [40.0 kB]
Get:14 http://apt.pop-os.org/ubuntu jammy-updates/universe Sources [272 kB]
Get:15 http://apt.pop-os.org/ubuntu jammy-updates/main amd64 DEP-11 Metadata [101 kB]
Get:16 http://apt.pop-os.org/ubuntu jammy-updates/universe amd64 Packages [984 kB]
Get:17 http://apt.pop-os.org/ubuntu jammy-updates/universe amd64 DEP-11 Metadata [289 kB]
Get:18 http://apt.pop-os.org/ubuntu jammy-updates/multiverse amd64 DEP-11 Metadata [940 B]
Get:19 http://apt.pop-os.org/ubuntu jammy-backports/main amd64 DEP-11 Metadata [4,896 B]
Get:20 http://apt.pop-os.org/ubuntu jammy-backports/universe amd64 DEP-11 Metadata [17.7 kB]
Fetched 3,734 kB in 1s (2,512 kB/s)                            
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
All packages are up to date.

**Comment by leviport** [comment]
Possibly a duplicate of https://github.com/pop-os/shop/issues/412

**Comment by PGQT** [comment]
My issue does not resolve if i wait. Although I have noticed it crashes
quicker if I click around on any tabs in the pop shop.

**Comment by jacobgkau** [comment]
@PGQT Does this crash still occur after installing the 7.3.0 update and rebooting? That update was released on 9/27/23 (last Wednesday).

**Comment by chloe-everhart** [comment]
I'm having this issue, too.  I just did sudo apt update && sudo apt upgrade.  It updated through the command line.  It still crashes within seconds after launch.

**Comment by jacobgkau** [comment]
> It still crashes within seconds after launch.

Does it happen every single time, or only sometimes? If it's every time, you could try getting a traceback by building and running a debug copy with these commands:

```
git clone https://github.com/pop-os/shop
cd shop
./rebuild.sh debug
run
```

The last command should happen when you see `(gdb)` at the beginning of the terminal prompt. After you do this, you'll want to `sudo apt remove pop-shop-dbgsym` to clean up (that package isn't in the repo, so it may cause errors trying to install updates if you have it, since apt will want to keep the main `pop-shop` package in sync with it.)

This may show us in more detail what's causing the segmentation fault that @PGQT pointed out.

**Comment by chloe-everhart** [comment]
>>
>> It still crashes within seconds after launch.
>>
>>
>
> Does it happen every single time, or only sometimes?
>
> Every time.
>  If it's every time, you could try getting a traceback by building and running a debug copy with these commands:
> git clone https://github.com/pop-os/shop./rebuild.sh debugrun
>
Here's what happened:

```
***@***.***:~$ git clone https://github.com/pop=os/shop./rebuild.sh debug
Cloning into 'debug'...
remote: Not Found
fatal: repository 'https://github.com/pop=os/shop./rebuild.sh/' not found
***@***.***:~$ sudo git clone https://github.com/pop-os/shop
[sudo] password for c: 
Cloning into 'shop'...
remote: Enumerating objects: 55019, done.
remote: Counting objects: 100% (4073/4073), done.
remote: Compressing objects: 100% (252/252), done.
Receiving objects:   8% (4402/55019), 964.00 Receiving objects:   9% (4952/55019), 964.00 Receiving objects:   9% (5328/55019), 1.75 MiReceiving objects:  10% (5502/55019), 1.75 MiReceiving objects:  10% (5782/55019), 2.12 MiReceiving objects:  11% (6053/55019), 3.02 MiReceiving objects:  11% (6248/55019), 3.02 MiReceiving objects:  12% (6603/55019), 3.70 MiReceiving objects:  13% (7153/55019), 4.54 MiReceiving objects:  13% (7240/55019), 4.54 MiReceiving objects:  14% (7703/55019), 5.64 MiReceiving objects:  14% (7727/55019), 5.64 MiReceiving objects:  15% (8253/55019), 6.32 MiReceiving objects:  15% (8635/55019), 6.32 MiReceiving objects:  16% (8804/55019), 6.88 MiReceiving objects:  17% (9354/55019), 6.88 MiReceiving objects:  17% (9594/55019), 7.61 MiReceiving objects:  18% (9904/55019), 8.50 MiReceiving objects:  19% (10454/55019), 9.14 MReceiving objects:  19% (10673/55019), 9.14 MReceiving objects:  20% (11004/55019), 10.00 Receiving objects:  20% (11164/55019), 10.00 Receiving objects:  21% (11554/55019), 10.60 Receiving objects:  22% (12105/55019), 10.60 Receiving objects:  23% (12655/55019), 11.36 Receiving objects:  23% (12657/55019), 11.36 Receiving objects:  24% (13205/55019), 12.66 Receiving objects:  24% (13378/55019), 12.66 Receiving objects:  25% (13755/55019), 13.16 Receiving objects:  25% (14164/55019), 13.61 Receiving objects:  26% (14305/55019), 14.01 Receiving objects:  27% (14856/55019), 14.60 Receiving objects:  27% (15287/55019), 14.60 Receiving objects:  28% (15406/55019), 14.60 Receiving objects:  29% (15956/55019), 15.35 Receiving objects:  30% (16506/55019), 15.35 Receiving objects:  30% (17044/55019), 16.13 Receiving objects:  31% (17056/55019), 16.13 Receiving objects:  32% (17607/55019), 16.89 Receiving objects:  32% (17921/55019), 16.89 Receiving objects:  33% (18157/55019), 16.89 Receiving objects:  34% (18707/55019), 17.40 Receiving objects:  35% (19257/55019), 17.40 Receiving objects:  36% (19807/55019), 18.09 Receiving objects:  36% (20032/55019), 18.09 Receiving objects:  37% (20358/55019), 18.95 Receiving objects:  38% (20908/55019), 19.50 Receiving objects:  38% (20944/55019), 19.50 Receiving objects:  39% (21458/55019), 19.82 Receiving objects:  40% (22008/55019), 20.22 Receiving objects:  40% (22155/55019), 20.22 Receiving objects:  41% (22558/55019), 20.22 Receiving objects:  42% (23108/55019), 20.89 Receiving objects:  42% (23428/55019), 21.64 Receiving objects:  43% (23659/55019), 21.64 Receiving objects:  44% (24209/55019), 22.29 Receiving objects:  45% (24759/55019), 22.29 Receiving objects:  45% (25265/55019), 22.96 Receiving objects:  46% (25309/55019), 22.96 Receiving objects:  47% (25859/55019), 23.41 Receiving objects:  48% (26410/55019), 23.86 Receiving objects:  48% (26501/55019), 23.86 Receiving objects:  49% (26960/55019), 23.86 Receiving objects:  50% (27510/55019), 24.50 Receiving objects:  50% (28017/55019), 25.04 Receiving objects:  51% (28060/55019), 25.04 Receiving objects:  52% (28610/55019), 25.54 Receiving objects:  53% (29161/55019), 25.54 Receiving objects:  53% (29564/55019), 26.20 Receiving objects:  54% (29711/55019), 26.20 Receiving objects:  55% (30261/55019), 26.87 Receiving objects:  56% (30811/55019), 27.37 Receiving objects:  56% (30829/55019), 27.37 Receiving objects:  57% (31361/55019), 27.85 Receiving objects:  58% (31912/55019), 28.46 Receiving objects:  58% (32004/55019), 28.46 Receiving objects:  59% (32462/55019), 28.46 Receiving objects:  60% (33012/55019), 29.16 Receiving objects:  61% (33562/55019), 29.16 Receiving objects:  62% (34112/55019), 29.16 Receiving objects:  62% (34407/55019), 29.64 Receiving objects:  63% (34662/55019), 29.64 Receiving objects:  64% (35213/55019), 29.64 Receiving objects:  65% (35763/55019), 30.30 Receiving objects:  66% (36313/55019), 30.30 Receiving objects:  67% (36863/55019), 30.30 Receiving objects:  67% (37278/55019), 30.95 Receiving objects:  68% (37413/55019), 30.95 Receiving objects:  69% (37964/55019), 30.95 Receiving objects:  70% (38514/55019), 30.95 Receiving objects:  71% (39064/55019), 30.95 Receiving objects:  72% (39614/55019), 31.45 Receiving objects:  73% (40164/55019), 31.45 Receiving objects:  74% (40715/55019), 31.45 Receiving objects:  74% (41119/55019), 31.89 Receiving objects:  75% (41265/55019), 31.89 Receiving objects:  76% (41815/55019), 31.89 Receiving objects:  77% (42365/55019), 31.89 Receiving objects:  78% (42915/55019), 32.44 Receiving objects:  79% (43466/55019), 33.18 Receiving objects:  79% (43474/55019), 33.18 Receiving objects:  80% (44016/55019), 33.18 Receiving objects:  81% (44566/55019), 33.65 Receiving objects:  81% (45054/55019), 34.27 Receiving objects:  82% (45116/55019), 34.27 Receiving objects:  83% (45666/55019), 34.27 Receiving objects:  84% (46216/55019), 34.84 Receiving objects:  85% (46767/55019), 34.84 Receiving objects:  85% (47129/55019), 35.54 Receiving objects:  86% (47317/55019), 35.54 Receiving objects:  87% (47867/55019), 35.54 Receiving objects:  88% (48417/55019), 36.21 Receiving objects:  89% (48967/55019), 36.21 Receiving objects:  89% (49006/55019), 36.68 Receiving objects:  89% (49493/55019), 37.51 Receiving objects:  90% (49518/55019), 37.51 Receiving objects:  91% (50068/55019), 37.51 Receiving objects:  92% (50618/55019), 37.51 Receiving objects:  93% (51168/55019), 37.98 Receiving objects:  94% (51718/55019), 37.98 Receiving objects:  94% (52200/55019), 37.98 Receiving objects:  95% (52269/55019), 38.63 Receiving objects:  96% (52819/55019), 39.44 Receiving objects:  96% (52891/55019), 39.44 Receiving objects:  97% (53369/55019), 40.50 Receiving objects:  97% (53466/55019), 40.50 Receiving objects:  98% (53919/55019), 41.07 Receiving objects:  99% (54469/55019), 41.66 Receiving objects:  99% (54826/55019), 41.66 remote: Total 55019 (delta 3855), reused 4006 (delta 3820), pack-reused 50946
Receiving objects: 100% (55019/55019), 42.31 Receiving objects: 100% (55019/55019), 42.63 MiB | 1.14 MiB/s, done.
Resolving deltas: 100% (50240/50240), done.
***@***.***:~$ ./rebuild.sh debug
bash: ./rebuild.sh: No such file or directory
***@***.***:~$ sudo ./rebuild.sh debug
sudo: ./rebuild.sh: command not found
***@***.***:~$ sudo git clone https://github.com/pop-os/shop ./rebuild.sh debug run
fatal: Too many arguments.

usage: git clone [<options>] [--] <repo> [<dir>]

    -v, --verbose         be more verbose
    -q, --quiet           be more quiet
    --progress            force progress reporting
    --reject-shallow      don't clone shallow repository
    -n, --no-checkout     don't create a checkout
    --bare                create a bare repository
    --mirror              create a mirror repository (implies bare)
    -l, --local           to clone from a local repository
    --no-hardlinks        don't use local hardlinks, always copy
    -s, --shared          setup as shared repository
    --recurse-submodules[=<pathspec>]
                          initialize submodules in the clone
    --recursive ...       alias of --recurse-submodules
    -j, --jobs <n>        number of submodules cloned in parallel
    --template <template-directory>
                          directory from which templates will be used
    --reference <repo>    reference repository
    --reference-if-able <repo>
                          reference repository
    --dissociate          use --reference only while cloning
    -o, --origin <name>   use <name> instead of 'origin' to track upstream
    -b, --branch <branch>
                          checkout <branch> instead of the remote's HEAD
    -u, --upload-pack <path>
                          path to git-upload-pack on the remote
    --depth <depth>       create a shallow clone of that depth
    --shallow-since <time>
                          create a shallow clone since a specific time
    --shallow-exclude <revision>
                          deepen history of shallow clone, excluding rev
    --single-branch       clone only one branch, HEAD or --branch
    --no-tags             don't clone any tags, and make later fetches not to follow them
    --shallow-submodules  any cloned submodules will be shallow
    --separate-git-dir <gitdir>
                          separate git dir from working tree
    -c, --config <key=value>
                          set config inside the new repository
    --server-option <server-specific>
                          option to transmit
    -4, --ipv4            use IPv4 addresses only
    -6, --ipv6            use IPv6 addresses only
    --filter <args>       object filtering
    --remote-submodules   any cloned submodules will use their remote-tracking branch
    --sparse              initialize sparse-checkout file to include only files at root

***@***.***:~$ ./rebuild.sh debug
bash: ./rebuild.sh: No such file or directory
***@***.***:~$ 
```

Then I tried to remove it:

```
***@***.***:~$ sudo apt remove pop-shop-dbgsym
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package pop-shop-dbgsym
***@***.***:~$ 
```

Did I do a command incorrectly?  What do I do from here?  Thank you for your help and trying to fix it.

**Comment by SRNissen** [comment]
> > It still crashes within seconds after launch.
> 
> Does it happen every single time, or only sometimes? If it's every time, you could try getting a traceback by building and running a debug copy with these commands:
> 
> ```
> git clone https://github.com/pop-os/shop
> ./rebuild.sh debug
> run
> ```

Pop!_Shop reliably crashes for me on certain actions so I attempted this build and it does not work.

On `./rebuild.sh debug`, I get this output:

```
+ killall io.elementary.appcenter
io.elementary.appcenter: no process found
+ true
++ dpkg-parsechangelog -S Version
+ VERSION=7.3.0pop0
+ rm -rf ./obj-x86_64-linux-gnu/
+ debuild -b -uc -us -nc
./rebuild.sh: line 8: debuild: command not found
```

I assume this requires `devscripts` to work? Or is it a different debuild that you use?

**Comment by jacobgkau** [comment]
> I assume this requires `devscripts` to work? Or is it a different debuild that you use?

Yes, install `devscripts` to use the `debuild` command (which that script does). You could also switch out `debuild` for `dpkg-buildpackage` (and the same flags) to not require `devscritps` (it just won't do some checks on the package, but those probably aren't needed for debugging, anyway).

**Comment by jacobgkau** [comment]
@chloe-everhart Your commands failed for multiple reasons:

- You typed `git clone https://github.com/pop=os/shop./rebuild.sh debug`, which is both multiple commands on the same line, and includes an equals sign where a hyphen should be in the GitHub URL.
- When you re-attempted the git command with the correct URL, you added `sudo` at the front (which nobody here recommended), which would give the files incorrect permissions and possibly put them in the wrong directory.

If you attempt it again, please log into GitHub to comment (rather than replying to the e-mails) so you can paste the output with proper formatting. Note that I added an additional step I left out the first time (changing into the `shop` directory after cloning it).

**Comment by chloe-everhart** [comment]
Thank you for the code to try.  I'm replying in email because I can't try it again, because after so long, I just ended up reinstalling that drive.  But truly, thanks for trying to help.

-- 
 Sent with Tuta; enjoy secure & ad-free emails: 
 https://tuta.com


Jan 26, 2024, 22:33 by ***@***.***:
